internal/servers/controller/handlers/sessions/session_service_test.go

```diff
@@ -292,6 +292,119 @@ func TestList(t *testing.T) {
+	sessRepo, err := session.NewRepository(rw, rw, kms)
+	require.NoError(t, err)
+
+	iamRepoFn := func() (*iam.Repository, error) {
+		return iamRepo, nil
+	}
+	sessRepoFn := func() (*session.Repository, error) {
+		return sessRepo, nil
+	}
+
+	_, pNoSessions := iam.TestScopes(t, iamRepo)
+	o, pWithSessions := iam.TestScopes(t, iamRepo)
+
+	at1 := authtoken.TestAuthToken(t, conn, kms, o.GetPublicId())
+	hc := static.TestCatalogs(t, conn, pWithSessions.GetPublicId(), 1)[0]
+	hs := static.TestSets(t, conn, hc.GetPublicId(), 1)[0]
+	h := static.TestHosts(t, conn, hc.GetPublicId(), 1)[0]
+	static.TestSetMembers(t, conn, hs.GetPublicId(), []*static.Host{h})
+	tar := target.TestTcpTarget(t, conn, pWithSessions.GetPublicId(), "test", target.WithHostSets([]string{hs.GetPublicId()}))
+
+	var wantSession []*pb.Session
+	for i := 0; i < 10; i++ {
+		sess := session.TestSession(t, conn, wrap, session.ComposedOf{
+			UserId:      at1.GetIamUserId(),
+			HostId:      h.GetPublicId(),
+			TargetId:    tar.GetPublicId(),
+			HostSetId:   hs.GetPublicId(),
+			AuthTokenId: at1.GetPublicId(),
+			ScopeId:     pWithSessions.GetPublicId(),
+			Endpoint:    "tcp://127.0.0.1:22",
+		})
+
+		status, states := convertStates(sess.States)
+
+		wantSession = append(wantSession, &pb.Session{
+			Id:                sess.GetPublicId(),
+			ScopeId:           pWithSessions.GetPublicId(),
+			AuthTokenId:       sess.AuthTokenId,
+			UserId:            sess.UserId,
+			TargetId:          sess.TargetId,
+			Endpoint:          sess.Endpoint,
+			HostSetId:         sess.HostSetId,
+			HostId:            sess.HostId,
+			Version:           sess.Version,
+			UpdatedTime:       sess.UpdateTime.GetTimestamp(),
+			CreatedTime:       sess.CreateTime.GetTimestamp(),
+			ExpirationTime:    sess.ExpirationTime.GetTimestamp(),
+			Scope:             &scopes.ScopeInfo{Id: pWithSessions.GetPublicId(), Type: scope.Project.String()},
+			Status:            status,
+			States:            states,
+			Certificate:       sess.Certificate,
+			Type:              target.TcpSubType.String(),
+			AuthorizedActions: []string{"read", "cancel"},
+		})
+	}
+
+	at2 := authtoken.TestAuthToken(t, conn, kms, o.GetPublicId())
+	_ = session.TestSession(t, conn, wrap, session.ComposedOf{
+		UserId:      at2.GetIamUserId(),
+		HostId:      h.GetPublicId(),
+		TargetId:    tar.GetPublicId(),
+		HostSetId:   hs.GetPublicId(),
+		AuthTokenId: at2.GetPublicId(),
+		ScopeId:     pWithSessions.GetPublicId(),
+		Endpoint:    "tcp://127.0.0.1:22",
+	})
+
+	cases := []struct {
+		name string
+		req  *pbs.ListSelfSessionsRequest
+		res  *pbs.ListSelfSessionsResponse
+		err  error
+	}{
+		{
+			name: "List Many Sessions",
+			req:  &pbs.ListSelfSessionsRequest{ScopeId: pWithSessions.GetPublicId()},
+			res:  &pbs.ListSelfSessionsResponse{Items: wantSession},
+		},
+		{
+			name: "List No Sessions",
+			req:  &pbs.ListSelfSessionsRequest{ScopeId: pNoSessions.GetPublicId()},
+			res:  &pbs.ListSelfSessionsResponse{},
+		},
```

Comment:
Can you make user parameterized, and then add a self lookup for user 2? You create a test session using `at2` above but then it's not actually used in the test.